lib/internal/bootstrap_node.js

```diff
@@ -377,8 +379,10 @@
       // that threw and was never cleared. So clear it now.
       async_id_fields[kInitTriggerAsyncId] = 0;
 
-      if (process.domain && process.domain._errorHandler)
-        caught = process.domain._errorHandler(er);
+      if (exceptionHandlerState.captureFn !== null) {
+        exceptionHandlerState.captureFn(er);
+        caught = true;
```

Comment:
I *think* this conflicts with `ClearFatalExceptionHandler` as it unsets `process.domain` but the `captureFn` would still exist and would throw again since `process.domain. _errorHandler` would be undefined. That won't be an issue once my PR for events lands but in the meantime, I think we need to update this code:

https://github.com/nodejs/node/blob/bb4462681cd2812d718897ca7ac8b8d48bca68f0/src/node.cc#L2479-L2482

We probably need better tests for that behaviour (I noticed it wasn't really tested while working on it myself)... unless I'm missing something with the above.

Reply:
@apapirovski Since you brought it up in the other PR: I’m writing a PR that removes `ClearFatalExceptionHandler` altogether :)

Reply:
@addaleax Sounds good! I guess it would either have to land before this or this would still need to be updated, right?

Out of curiosity, what direction are you going with it? (Because technically mine removed it too but I'm guessing you have a different solution in mind.)

Reply:
@apapirovski Here’s the PR: https://github.com/nodejs/node/pull/17333 :)

Reply:
Oh, and regarding updating this PR: The situation you’re worried about is a domain setting a capture function, that would proceed to capture the exception generated by one of these?

I *think* you are right, #17333 should probably land first (or at least this should not land without it)…